mission-control/docs/config-db/concepts/custom-scrapers.md

```diff
@@ -0,0 +1,89 @@
+
+```yaml title="file-scraper.yaml"
+apiVersion: configs.flanksource.com/v1
+kind: ScrapeConfig
+metadata:
+  name: file-scraper
+spec:
+  file:
+    - type: $.Config.InstanceType
+      class: $.Config.InstanceType
+      id: $.Config.InstanceId
+      path:
+        - config*.json
+        - test*.json
+```
+
+## Mapping
+
+Custom scrapers need to define the id, type & class for each items that are scraped. For example: if you're scraping a file with a JSON Array where each element on the array is to be scraped as a config item, you need to define what the id, type & config class of the items should be. This can be done using mappings.
+
+| Field             | Description                                                                      | Scheme   | Required |
+| ----------------- | -------------------------------------------------------------------------------- | -------- | -------- |
+| `id`              | Specify JSONPath to get the id for each config item                              | `string` | `true`   |
+| `type`            | Specify JSONPath to get the type for each config item                            | `string` | `true`   |
+| `class`           | Specify JSONPath to get the class for each config item. _(Defaults to `type`)_   | `string` |          |
+| `name`            | Specify JSONPath to get the name for each config item                            | `string` |          |
+| `createFields`    | Specify JSONPath to get the creation timestamp for each config item              | `string` |          |
+| `deleteFields`    | Specify JSONPath to get the deletion timestamp for each config item              | `string` |          |
+| `timestampFormat` | Specify the format of the timestamp specified in `createFields` & `deleteFields` | `string` |          |
+
+### Date Mapping
+
+Date mapping allows you to specify custom creation and deletion times for config items. This is useful when you want to import config items from an external source, and you want to preserve the creation and deletion times of the config items in the external source.
+
+You'll be making use of the `createFields` and `deleteFields` fields. They are both a list of JSONPath expression and are used to extract the created/deleted time of the config item from the scraped configuration. If multiple fields are specified, the first non-empty value will be used.
+
+Consider the following configuration file
+
+```yaml title="kubernetes-scraper.yaml"
+apiVersion: configs.flanksource.com/v1
+kind: ScrapeConfig
+metadata:
+  name: kubernetes-scraper
+spec:
+  file:
+    - type: $.aws[0].region
+      id: $.aws[0].region
+      createFields:
+        - $.aws[0].made_at
+        - $.aws[0].created_at
+      deleteFields:
+        - '$.aws[0].removed_at'
+        - '$.aws[0].deleted_at'
+      paths:
+        - fixtures/data/test.yaml
+```
+
+where `fixtures/data/test.yaml` is
+
+```yaml title="aws-scraper.yaml"
+apiVersion: configs.flanksource.com/v1
```

Comment:
Use a different source file as example - its confusing with a CRD

Reply:
We have a tutorial that works with git source